### DISCUSSION TITLE
How to configure S3 gateway port for s3g

### DISCUSSION THREAD

**Discussion by julienlau** [apache/ozone#7502] [FAQ] [answered]
Hi,
s3g default http port is 9878.
How can I change it to something else ?
I don't think it is supported yet when I look at  [the code](https://github.com/apache/ozone/blob/9b261560d651a39875026da6bbf5364b7dbd50a4/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/S3GatewayHttpServer.java#L136) 

Regards

**Comment by adoroszlai** [comment, marked as answer]
Set `ozone.s3g.http-address` and/or `ozone.s3g.https-address`.

https://github.com/apache/ozone/blob/e4b6007c8e9f79d1432cf6f128d6f83e733f8e9b/hadoop-hdds/common/src/main/resources/ozone-default.xml#L1791-L1798

https://github.com/apache/ozone/blob/e4b6007c8e9f79d1432cf6f128d6f83e733f8e9b/hadoop-hdds/common/src/main/resources/ozone-default.xml#L1818-L1824

Which one of the two is used depends on:

https://github.com/apache/ozone/blob/e4b6007c8e9f79d1432cf6f128d6f83e733f8e9b/hadoop-hdds/common/src/main/resources/ozone-default.xml#L3261-L3271

**Reply by julienlau** [reply]
thank you very much
